### PR TITLE
Bug fix 3.5/add db list transactions function

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Added missing function `db._transactions()` plus REST route GET /_api/transaction

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behaviour change can only be verified via automatic tests

#### Related Information

- [x] There is an internal planning ticket: https://github.com/arangodb/release-3.5/issues/83

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5434/

### CODE REVIEW

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5439/

File: js/client/modules/@arangodb/arango-transaction.js [hunk 1]
```diff
@@ -129,18 +161,38 @@ ArangoTransaction.prototype.commit = function() {
...
+  arangosh.checkRequestResult(requestResult);
+  return requestResult.result;
+};
 
+ArangoTransaction.prototype.running = function() {
+  return this._id !== 0 && !this._done;
+};
+
+ArangoTransaction.prototype.query = function(query, bindVars, cursorOptions, options) {
+  if (!this.running()) {
```
Comment (graetzer) [inline js/client/modules/@arangodb/arango-transaction.js]: use `throwNotRunning();` ?

Comment (jsteemann) [reply js/client/modules/@arangodb/arango-transaction.js]: good point

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5443/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**